crates/bevy_winit/src/lib.rs

```diff
@@ -703,19 +722,20 @@ pub fn winit_runner(mut app: App) {
                 });
             }
             event::Event::Suspended => {
-                runner_state.is_active = false;
-                #[cfg(target_os = "android")]
-                {
-                    // Remove the `RawHandleWrapper` from the primary window.
-                    // This will trigger the surface destruction.
-                    let mut query = app.world.query_filtered::<Entity, With<PrimaryWindow>>();
-                    let entity = query.single(&app.world);
-                    app.world.entity_mut(entity).remove::<RawHandleWrapper>();
-                    *control_flow = ControlFlow::Wait;
-                }
+                let (mut event_writers, _, _) = event_writer_system_state.get_mut(&mut app.world);
+                event_writers.lifetime.send(Lifetime::Suspended);
+                // Mark the state as `WillSuspend`. This will let the schedule run one last time
+                // before actually suspending to let the application react
+                runner_state.active = ActiveState::WillSuspend;
             }
             event::Event::Resumed => {
-                runner_state.is_active = true;
+                let (mut event_writers, _, _) = event_writer_system_state.get_mut(&mut app.world);
+                match runner_state.active {
+                    ActiveState::NotYetStarted => event_writers.lifetime.send(Lifetime::Started),
+                    ActiveState::Suspended => event_writers.lifetime.send(Lifetime::Resumed),
+                    _ => unreachable!(),
```

Comment:
Are we _really_ sure it's impossible to get a `Resumed` event while we're in `WillSuspend` and `Active`?

Reply:
yes, but we can play it safe